types/src/contracts.rs

```diff
@@ -548,6 +548,50 @@ impl JsonSchema for ContractPackageHash {
     }
 }
 
+/// A enum to determine the lock status of the contract package.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+pub enum ContractPackageStatus {
+    /// The package is locked and cannot be versioned.
+    Locked,
+    /// The package is unlocked and can be versioned.
+    Unlocked,
+}
+
+impl Default for ContractPackageStatus {
+    fn default() -> Self {
+        Self::Unlocked
+    }
+}
+
+impl ToBytes for ContractPackageStatus {
+    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
+        let mut result = bytesrepr::allocate_buffer(self)?;
+        match self {
+            ContractPackageStatus::Unlocked => result.append(&mut false.to_bytes()?),
+            ContractPackageStatus::Locked => result.append(&mut true.to_bytes()?),
+        }
+        Ok(result)
+    }
+
+    fn serialized_length(&self) -> usize {
+        match self {
+            ContractPackageStatus::Unlocked => false.serialized_length(),
+            ContractPackageStatus::Locked => true.serialized_length(),
+        }
+    }
+}
+
+impl FromBytes for ContractPackageStatus {
+    fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), bytesrepr::Error> {
+        let (val, bytes) = bool::from_bytes(bytes)?;
+        if val {
```

Comment:
This also could benefit from a special constructor that would hide this condition

Reply:
Added a `ContractPackageStatus::new` to help code simplification

Reply:
Used a special constructor